app/models/event.rb

```diff
@@ -24,4 +22,17 @@ class Event < ApplicationRecord
   def service_description
     service_category.service_category_name
   end
+
+  def exception_notes
```

Comment:
I like that you are using an instance method instead of a class method.  I did notice an increase in response time with this PR (about 60 milliseconds average over 1,000 iterations using 43123 which has27 events).  Is there potential in improving performance using a join instead of stepping through each record?

Reply:
Yeah I think there is, but it should come from the controller. The model should only be concerned with its associations. The object that's doing the querying (in this case the controller) should worry about the eager loading of objects. If the objects are already loaded then step through them should be performant. It's the repeated querying that's slowing us down (I think).
I'm pretty sure we're on the same page about this, but the performance improvements make a lot of sense as their own pr so that we can benchmark and test them in isolation.

Reply:
I rewrote exception_notes in the event model like so:


def exception_notes
&nbsp;&nbsp;event_zip_codes.joins(:event_geography)
 &nbsp;&nbsp;.where(event_id: self.event_id, trigger_exception_note: 1 )
  &nbsp;&nbsp;.pluck(:zip_code, :exception_note).to_h
  end


It doesn't rely entirely on associations.  Perhaps I am not using them correctly.  But it shaves an average of 100 ms off of the average benchmark.  It doesn't put the exception_notes in an array but when talking to David about that earlier he said that wouldn't happen unless by mistake.